src/main/java/mekanism/common/content/gear/mekasuit/ModuleMekaSuit.java

```diff
@@ -45,15 +44,21 @@
 
         @Override
         public void tickServer(PlayerEntity player) {
-            if (player.func_233571_b_(FluidTags.WATER) > 1.8) {
+            if (player.func_233571_b_(FluidTags.WATER) > 1.6) {
                 FloatingLong usage = MekanismConfig.general.FROM_H2.get().multiply(2);
                 long maxRate = Math.min(getMaxRate(), getContainerEnergy().divide(usage).intValue());
                 long hydrogenUsed = 0;
                 GasStack hydrogenStack = new GasStack(MekanismGases.HYDROGEN.get(), maxRate * 2);
                 ItemStack chestStack = player.getItemStackFromSlot(EquipmentSlotType.CHEST);
-                Optional<IGasHandler> capability = MekanismUtils.toOptional(chestStack.getCapability(Capabilities.GAS_HANDLER_CAPABILITY));
-                if (Modules.load(chestStack, Modules.JETPACK_UNIT) != null && capability.isPresent()) {
-                    hydrogenUsed = maxRate * 2 - capability.get().insertChemical(hydrogenStack, Action.EXECUTE).getAmount();
+                Optional<IGasHandler> chestCapability = MekanismUtils.toOptional(chestStack.getCapability(Capabilities.GAS_HANDLER_CAPABILITY));
+                if (chestCapability.isPresent()) {
+                    hydrogenUsed = maxRate * 2 - chestCapability.get().insertChemical(hydrogenStack, Action.EXECUTE).getAmount();
+                }
+                hydrogenStack.setAmount(maxRate * 2 - hydrogenUsed);
```

Comment:
Moving this in the above if statement and changing it to `hydrogenStack.shrinkStack(hydrogenUsed)` may make this easier to see what is going on